src/main/java/me/pheric/pcore/database/Database.java

```diff
@@ -0,0 +1,171 @@
+package me.pheric.pcore.database;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Manages and connects to a MySQL database
+ *
+ * @author StealthMajr (stealthmajr@gmail.com)
+ * @see Connection
+ * @see HikariDataSource
+ * @see Result
+ * @see DB
+ * @since 0.1.0
+ */
+public class Database {
+
+    private HikariDataSource dataSource;
+
+    /**
+     * Initializes the class
+     *
+     * @param jdbcUrl  The JDBC URL for connecting, usually in the following format:
+     *                 jdbc:mysql://hostname:port/database
+     * @param username The username of this MySQL connection
+     * @param password The password of this MySQL connection
+     */
+    public Database(String jdbcUrl, String username, String password) {
+        dataSource = new HikariDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
+        dataSource.setJdbcUrl(jdbcUrl);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        dataSource.setPoolName("Core");
```

Comment:
Pool name will be "Core" even if another plugin is using the pool.

Reply:
In my case it was a core-only scenario, I'll fix. Thanks!